src/main/java/io/confluent/idesidecar/restapi/credentials/Redactable.java

```diff
@@ -42,6 +45,19 @@ public T deserialize(JsonParser p, DeserializationContext ctxt) throws IOExcepti
     protected abstract T create(char[] value);
   }
 
+  protected abstract static class BaseSerializer<T extends Redactable> extends JsonSerializer<T> {
```

Comment:
We should call this `RawSerializer` or something, to make it more obvious this will serialize the **raw value of the `Redactable`**, which is unsafe to use in production.

In fact, we might want to move all this code to the `SidecarClient` class, so that we never have a serializer for `Redactable` subclasses in production code.